addons/account_edi/models/account_journal.py

```diff
@@ -10,7 +11,40 @@ class AccountJournal(models.Model):
     edi_format_ids = fields.Many2many(comodel_name='account.edi.format',
                                       string='Electronic invoicing',
                                       help='Send XML/EDI invoices',
-                                      domain="[('hide_on_journal', '=', 'import_export')]")
+                                      domain="[('id', 'in', compatible_edi_ids)]",
+                                      compute='_compute_edi_format_ids',
+                                      readonly=False, store=True)
+
+    compatible_edi_ids = fields.Many2many(comodel_name='account.edi.format', relation='account_journal_compatible_account_edi_format',
+                                          compute='_compute_compatible_edi_ids')
+
+    def write(self, vals):
+        old_edi_format_ids = self.edi_format_ids
+        super().write(vals)
+        diff_edi_format_ids = old_edi_format_ids - self.edi_format_ids
+        documents = self.env['account.edi.document'].search([('edi_format_id', 'in', diff_edi_format_ids.ids), ('state', 'in', ('to_post', 'to_send'))])
+        if documents:
+            raise UserError(_('Cannot deactivate (%s) on this journal because not all documents are synchronized', ', '.join([f.name for f in documents.edi_format_id])))
+
+    @api.depends('type', 'company_id', 'company_id.country_id')
+    def _compute_compatible_edi_ids(self):
+        """ Is this EDI compatible on the journal ?
```

Comment:
I added a helper on the field, but I'm not sure it's very explicit. I don't what to put because the field name seems pretty self-explanatory to me (EDI format compatible with the journal).